src/Engine/ProtoCore/Namespace/ElementRewriter.cs

```diff
@@ -0,0 +1,213 @@
+﻿using System.Diagnostics;
+using ProtoCore.AST;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.DSASM;
+using ProtoCore.Utils;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace ProtoCore.Namespace
+{
+
+    public class ElementRewriter
+    {
+        private readonly ElementResolver elementResolver;
+
+        private ElementRewriter(ElementResolver elementResolver)
+        {
+            this.elementResolver = elementResolver;
+        }
+
+        /// <summary>
+        /// Lookup namespace resolution map to substitute 
+        /// partial classnames with their fully qualified names in ASTs.
+        /// If partial class is not found in map, 
+        /// update ResolutionMap with fully resolved name from compiler.
+        /// </summary>
+        /// <param name="classTable"></param>
+        /// <param name="elementResolver"></param>
+        /// <param name="astNodes"> parent AST node </param>
+        public static void ReplaceClassNamesWithResolvedNames(ClassTable classTable,
+            ElementResolver elementResolver, ref IEnumerable<Node> astNodes)
```

Comment:
With these arguments being self-explanatory, I'd much rather the method be renamed (note the sequence of arguments and the removal of `ref`):

```csharp
public static void Rewrite(ElementResolver resolver, ClassTable classTable, IEnumerable<Node> astNodes)
{
    // ...
}
```